service/history/resource/resource.go

```diff
@@ -147,10 +156,36 @@ func New(
 		return nil, fmt.Errorf("invalid ratelimit algorithm config: %w", err)
 	}
 
+	archivalClient := archiver.NewClient(
```

Comment:
archiverClient is only used by history & worker but resource is shared across all services. It's probably negligible from resource usage/side effect perspective to initialize it here but ideally we can change this to lazy init.

Reply:
oh, this resource is only for history service, it's inside `service/history/resource` package, not inside `common/resource` package.